pallets/order-book/src/lib.rs

```diff
@@ -525,464 +484,346 @@ pub mod pallet {
 		#[pallet::weight(T::Weights::rm_trading_pair())]
 		pub fn rm_trading_pair(
 			origin: OriginFor<T>,
-			asset_in: T::AssetCurrencyId,
-			asset_out: T::AssetCurrencyId,
+			currency_in: T::CurrencyId,
+			currency_out: T::CurrencyId,
 		) -> DispatchResult {
 			T::AdminOrigin::ensure_origin(origin)?;
 
 			// We do not check, we just remove as this is an admin action.
-			TradingPair::<T>::remove(asset_in, asset_out);
+			TradingPair::<T>::remove(currency_in, currency_out);
 
 			Self::deposit_event(Event::<T>::TradingPairRemoved {
-				asset_in,
-				asset_out,
+				currency_in,
+				currency_out,
 			});
 
 			Ok(())
 		}
 
-		/// Sets the minimum order amount for a given trading pair.
-		/// If the trading pair is not yet added this errors out.
-		//
-		// NOTE: We do not need to update any existing orders as fulfillment does
-		//       not verify the validity of the order that is to be fulfilled.
+		/// Set the market feeder for set market ratios.
+		/// The origin must be the admin origin.
 		#[pallet::call_index(6)]
-		#[pallet::weight(T::Weights::update_min_order())]
-		pub fn update_min_order(
-			origin: OriginFor<T>,
-			asset_in: T::AssetCurrencyId,
-			asset_out: T::AssetCurrencyId,
-			min_order: T::Balance,
-		) -> DispatchResult {
+		#[pallet::weight(T::Weights::set_market_feeder())]
+		pub fn set_market_feeder(origin: OriginFor<T>, feeder_id: T::FeederId) -> DispatchResult {
 			T::AdminOrigin::ensure_origin(origin)?;
 
-			// try_mutate is a pain with the direct error return. But we do want to only
-			// update if the pair exists.
-			let _old_min_order = TradingPair::<T>::get(&asset_in, &asset_out)?;
-			TradingPair::<T>::insert(&asset_in, &asset_out, min_order);
+			MarketFeederId::<T>::put(feeder_id);
```

Comment:
Events are a means of saying that the state changed, so yes indexers will most likely only see that change if there was an evenrt.

Reply:
Ok, because the indexer can not also index "calls" done, right? Because in that case, just seeing that the call has been performed, they now the state has changed

Reply:
Done anyways here: [4a0f52a](https://github.com/centrifuge/centrifuge-chain/pull/1674/commits/4a0f52ab67e899da74e4b504a73020f63616c255)